### PR TITLE
Issue #153 Added attributes for sheet definitions to customise preview image more 

### PR DESCRIPTION
**Pull request by ReneeIbeji**
Added 3 attributes that can be used in the items sheet definitions :
- preview_column - Represents which sprite on a row to use as the preview image (default 0)
- preview_x_offset - The amount of pixels on the x axis the preview image frame is offsetted by in reference to the preview column set. (default 0)
- preview_y_offset - The amount of pixels on the y axis the preview image frame is offsetted by in reference to the preview row set.
(default 0)

Please note the default value for the preview row is 10.
I've used these new attributes to fix the preview images for all the spear items so far.

### CODE REVIEW

**Comment by jrconway3** [comment]
I'll try testing this out later to see if I can get it working.

**Comment by jrconway3** [comment]
Finally got a chance to look, forgot about this before. Looking pretty good! Its really great being able to see weapon previews finally. Sucks we have to manually make changes to showcase the correct preview, but at least the option is there. It also lets us pick out a better preview it a different preview might look better, too.

**Review by sanderfrenken — APPROVED** [review]

**Comment by sanderfrenken** [comment]
I see you need to rebase, there are some conflicts. Thanks for the contribution already!

File: sources/chargen.js [hunk 1]
```diff
@@ -665,27 +667,33 @@ $(document).ready(function() {
...
-          }
-        }    
+        const animation =  $(this).data(`layer_1_custom_animation`);
+
+        if($(this).data(`layer_1_${getBodyTypeName()}`) === undefined){
+          previewToDraw.link = $(this).data(`layer_1_${getBodyTypeName()}`);
+          previewToDraw.zPos = $(this).data(`layer_1_zpos`);
+          layers.push(previewToDraw);
+        }
+        else{
```
Comment (sanderfrenken) [inline sources/chargen.js]: could you move this else on the closing bracket of the above if?

Comment (ReneeIbeji) [reply sources/chargen.js]: I think I fixed it

**Comment by ReneeIbeji** [comment]
OK, I don't think their should be any issues now

**Pull request merged**